.github/workflows/test.yaml

```
@@ -14,14 +14,28 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Install Node.js
-        uses: useblacksmith/setup-node@v5
+      - name: Setup tools
+        uses: ./.github/actions/setup-tools
         with:
-          node-version: 18.x
-          cache: 'npm'
+          install-nodejs: 'true'
 
-      - name: Install Dependencies
-        run: npm ci
+      - name: Build project and capture output
+        run: npm run build > build.log 2>&1
+
+      - name: Check for build errors
+        run: |
+          if grep -q -i "error" build.log || grep -q -i "warning" build.log; then
+            echo "Build contains following errors or warnings..."
+
+            cat build.log
+
+            exit 1;
+          else
+            exit 0;
+          fi
+
+      - name: Set environment variable
+        run: export CHECK_CONTRACT_SIZE=true
 
       - name: Test
```

```suggestion
      - name: Test
        env:
          CHECK_CONTRACT_SIZE=true
```
the exported variable won't be available in the next step, right? Also, the env var only needs to be defined for this command, so prefer to not create a separate step